### PR TITLE
Fixed Xcode 13.2.x / Swift 5.5 compatibility

### PR DESCRIPTION
**Pull request by NachoSoto**
Fixes [SDKONCALL-156].

This will be covered by #1990.

[SDKONCALL-156]: https://revenuecats.atlassian.net/browse/SDKONCALL-156?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Sources/Identity/IdentityManager.swift [hunk 1]
```diff
@@ -110,7 +110,7 @@ private extension IdentityManager {
         guard newAppUserID != currentAppUserID else {
             Logger.warn(Strings.identity.logging_in_with_same_appuserid)
             self.customerInfoManager.customerInfo(appUserID: currentAppUserID,
-                                                  fetchPolicy: .cachedOrFetched) { result in
+                                                  fetchPolicy: .cachedOrFetched) { @Sendable result in
```
Comment (aboedo) [inline Sources/Identity/IdentityManager.swift:113]: were these implicitly sendable in more recent Xcodes?

File: Sources/Purchasing/Purchases/PurchasesOrchestrator.swift [hunk 2]
```diff
@@ -914,10 +914,18 @@ private extension PurchasesOrchestrator {
...
 
+                #if swift(>=5.6)
                 @MainActor
                 func complete() {
                     completion?(transaction, nil, purchasesError, false)
                 }
+                #else
+                @Sendable
+                @MainActor
+                func complete() {
```
Comment (aboedo) [inline Sources/Purchasing/Purchases/PurchasesOrchestrator.swift:925]: can't wait until April when Xcode 13 is dropped and we can come back to kill this bit

**Review by aboedo — APPROVED** [review]

File: Sources/Networking/Caching/CustomerInfoCallback.swift [hunk 3]
```diff
@@ -15,7 +15,7 @@ import Foundation
 
 struct CustomerInfoCallback: CacheKeyProviding {
 
-    typealias Completion = (Result<CustomerInfo, BackendError>) -> Void
+    typealias Completion = @Sendable (Result<CustomerInfo, BackendError>) -> Void
```
Comment (aboedo) [inline Sources/Networking/Caching/CustomerInfoCallback.swift:18]: this is what the tests are complaining about, right? 
I don't know how to solve it cleanly, though - it looks like Xcode 14 doesn't like the idea of mixing concurrency stuff with non-concurrency

File: Sources/Networking/CustomerAPI.swift [hunk 4]
```diff
@@ -119,5 +119,3 @@ final class CustomerAPI {
     }
 
 }
-
-extension CustomerAPI: Sendable {}
```
Comment (NachoSoto) [inline Sources/Networking/CustomerAPI.swift:123]: This was unnecessary and fixed a lot of unnecessary warnings.

File: Sources/Networking/Caching/CustomerInfoCallback.swift (hunk 3 above)
Comment (NachoSoto) [reply Sources/Networking/Caching/CustomerInfoCallback.swift]: Removed!

**Review by aboedo — APPROVED** [review]

**Pull request merged**